### ISSUE TITLE
Task List

### ISSUE THREAD

**Issue by sam99dave** [sam99dave/profiling-and-optimizations#1]
- [x] PyTorch Benchmark ( [link](https://pytorch.org/tutorials/recipes/recipes/benchmark.html) )
- [x] Python Threading
- [x] Profiler with TensorBoard ( [link](https://pytorch.org/tutorials/intermediate/tensorboard_profiler_tutorial.html) )
- [x] Triton-kernels with torch.compile ( [link](https://pytorch.org/tutorials/recipes/torch_compile_user_defined_triton_kernel_tutorial.html) )
- [ ] [Large Scale Transformer model training with Tensor Parallel (TP)](https://pytorch.org/tutorials/intermediate/TP_tutorial.html)
- [ ] [Accelerating BERT with semi-structured (2:4) sparsity](https://pytorch.org/tutorials/advanced/semi_structured_sparse.html)
- [ ] [torch.export Tutorial with torch.export.Dim](https://pytorch.org/tutorials/intermediate/torch_export_tutorial.html)
